src/core/Sizeable.js

```diff
@@ -131,14 +175,14 @@ const SizeableMixin = base => {
                 )
             }
             else if (newValue instanceof Array) {
-                if (typeof newValue[0] != 'undefined') { this._properties[name]._x = newValue[0]; change = true }
-                if (typeof newValue[1] != 'undefined') { this._properties[name]._y = newValue[1]; change = true }
-                if (typeof newValue[2] != 'undefined') { this._properties[name]._z = newValue[2]; change = true }
+                if (typeof newValue[0] != 'undefined') { this._properties[name].x = newValue[0]; change = true }
+                if (typeof newValue[1] != 'undefined') { this._properties[name].y = newValue[1]; change = true }
+                if (typeof newValue[2] != 'undefined') { this._properties[name].z = newValue[2]; change = true }
             }
             else {
-                if (typeof newValue.x != 'undefined') { this._properties[name]._x = newValue.x; change = true }
-                if (typeof newValue.y != 'undefined') { this._properties[name]._y = newValue.y; change = true }
-                if (typeof newValue.z != 'undefined') { this._properties[name]._z = newValue.z; change = true }
+                if (typeof newValue.x != 'undefined') { this._properties[name].x = newValue.x; change = true }
+                if (typeof newValue.y != 'undefined') { this._properties[name].y = newValue.y; change = true }
+                if (typeof newValue.z != 'undefined') { this._properties[name].z = newValue.z; change = true }
```

Comment:
Looks good. Now that we're using the `x`, `y`, `z` getters, they trigger a private `valuechanged` event which causes `ImperativeBase` to trigger a public `propertychanged` event that someone using the API can react to.

However, this method here also triggers a public `propertychange` event (public and private events need to be documented at some point once we have a documentation site). This means that two `propertychange` events will be triggered.

How can we fix this issue, so that two `propertychange` events are not triggered?

My first thought it to just remove the following line that triggers the `propertychange` event, and let the now-used setters trigger the event.

(sidenote: if we're going to have an event system for this, we should be more fined tuned, and perhaps trigger events like `valuechanged.x`, `valuechanged.y`, and `valuechanged.z` rather than just a `valuechanged` event which currently will make people have to write conditional checks to see which property has changed.

Reply:
So you're thinking to move the line "if (change) this.triggerEvent('propertychange', name)"?

Reply:
to remove, yes, I think that'll be good enough for this PR, and that we should make it better in #140

Reply:
It's because if you look on these lines, https://github.com/trusktr/infamous/pull/139/files#diff-ba5f48e01244e9fef97bad2bc4b08dcaL48 , you'll see that it already observes `valuechanged`, which is triggered by using the x,y,z setters.